### ISSUE TITLE
[fluent-react] Localized nested elements as props is failing 

### ISSUE THREAD

**Issue by hkasemir** [projectfluent/fluent.js#89]
I have the following (simplified, I know the link should also be localized, but first things first) code: 

```js
<Localized
  id='door-code-explanation'
  $contactLink={<a href='mailto:support@foo.com'>Contact us</a>}>
  <p>
    This code will be surfaced to people attending the upcoming session.
  </p>
</Localized>
```

```
// ftl
door-code-explanation = 
    This code will be surfaced to people attending the upcoming session.
    { $contactLink } if you have further questions.
```

what I would expect to see:
> This code will be surfaced to people attending the upcoming session. [Contact us](foo.com) if you have further questions.

instead I see
> This code will be surfaced to people attending the upcoming session. contactLink if you have further questions.

There are no errors in the console.
Also, when I change the `$contactLink` prop to a regular string, the result is as expected.

**Comment by stasm** [comment]
Thanks for the report. This should work in `fluent-react` 0.4.x. I'll take a look. I see that there are no tests for this so perhaps I regressed something.

**Comment by stasm** [comment]
It looks like this is Webpack's fault. At least in my setup which uses `create-react-app`.

When Webpack bundles all modules and their dependencies, it ends up duplicating the following imports (also when I `import from 'fluent'` rather than `import from 'fluent/compat'`). `FluentType` should `=== MessageArgument` and yet it doesn't.

https://github.com/projectfluent/fluent.js/blob/cce4d42997e33cb8d0960bd62a56f5db36ca190a/fluent/src/resolver.js#L50

https://github.com/projectfluent/fluent.js/blob/cce4d42997e33cb8d0960bd62a56f5db36ca190a/fluent-react/src/localized.js#L3

I think the solution is to do what React does with its elements: add a `$$typeof: Symbol.for('FluentType')` field to all `FluentTypes` so that it's easy to discover them without relying on `instanceof`.

That said, my mid-term plan is to remove `MessageContext.formatToParts` and remove the feature of passing React elements as arguments to translations and replace it with a more declarative approach, similar to DOM overlays in `fluent-web`.

**Comment by hkasemir** [comment]
OK, I know that for us it is important to be able to include components or DOM elements as jsx. I don't know what you mean by DOM overlays - do you have examples or documentation?

**Comment by stasm** [comment]
I just published `fluent` 0.4.2 which fixed this.
